test/unit/specs/components/staking/PageValidator.spec.js

```diff
@@ -251,107 +253,209 @@ describe(`onStake`, () => {
-        commit: jest.fn(),
-        dispatch: jest.fn(() => {
-          throw new Error(`one\ntwo\nthree\nfour\nfive\nsix"seven`)
-        }),
-        getters: getterValues
-      }
+        await submitDelegation({ amount: 10 })
+
+        expect($store.dispatch.mock.calls).toEqual([
+          [`submitDelegation`, [{ atoms: BigNumber(10), delegate }]]
+        ])
+
+        expect($store.commit.mock.calls).toEqual([
+          [
+            `notifyError`,
+            {
+              body: `message`,
+              title: `Error While Staking atoms`
+            }
+          ]
+        ])
+      })
 
-      const {
-        vm: { submitDelegation }
-      } = mount(PageValidator, {
-        mocks: {
-          $route: { params: { validator: `1a2b3c` } },
-          $store
+      it(`error with data`, async () => {
+        const $store = {
+          commit: jest.fn(),
+          dispatch: jest.fn(() => {
+            throw new Error(`one\ntwo\nthree\nfour\nfive\nsix"seven`)
+          }),
+          getters: getterValues
         }
+
+        const {
+          vm: { submitDelegation }
+        } = mount(PageValidator, {
+          mocks: {
+            $route: { params: { validator: `1a2b3c` } },
+            $store
+          }
+        })
+
+        await submitDelegation({ amount: 10 })
+
+        expect($store.dispatch.mock.calls).toEqual([
+          [`submitDelegation`, [{ atoms: BigNumber(10), delegate }]]
+        ])
+
+        expect($store.commit.mock.calls).toEqual([
+          [
+            `notifyError`,
+            {
+              body: `Seven`,
+              title: `Error While Staking atoms`
+            }
+          ]
+        ])
       })
+    })
+
+    describe(`composition`, () => {
+      it(`delegation.submitDelegation`, async () => {
+        const delegation = Delegation({})
 
-      await submitDelegation({ amount: 10 })
+        const dispatch = jest.fn((type, payload) => {
+          if (type === `submitDelegation`) {
+            delegation.actions[type]($store, payload)
+          }
+        })
 
-      expect($store.dispatch.mock.calls).toEqual([
-        [`submitDelegation`, [{ atoms: BigNumber(10), delegate }]]
-      ])
+        const $store = {
+          commit: jest.fn(),
+          dispatch,
+          getters: getterValues,
+          rootState: getterValues,
+          state: {
+            committedDelegates: { "1a2b3c": 0 },
+            unbondingDelegations: {}
+          }
+        }
 
-      expect($store.commit.mock.calls).toEqual([
-        [
-          `notifyError`,
-          {
-            body: `Seven`,
-            title: `Error While Staking atoms`
+        const {
+          vm: { submitDelegation }
+        } = mount(PageValidator, {
+          mocks: {
+            $route: { params: { validator: `1a2b3c` } },
+            $store
           }
-        ]
-      ])
+        })
+
+        await submitDelegation({ amount: 10 })
+
+        expect($store.dispatch.mock.calls).toEqual([
```

Comment:
I think snapshots are a bad idea for these reasons:

* They are brittle.
* They don't express customer value well.
* They're stored in a separate file so you can't see what the test is expecting easily.
* They require an external tool to manipulate (instead of just your code editor).

Reply:
> They're stored in a separate file so you can't see what the test is expecting easily

true.

> They require an external tool to manipulate (instead of just your code editor)

false.

I agree with `simpler = better`. But it is a lot more comfortable to update snapshots then go through code on a format change and change all the expected values. Let's discuss it.

Reply:
I know we use snapshots to test HTML structure elsewhere.  Is that what you mean by "format change"?  The code above isn't testing HTML—it's testing passed arguments to a called function.

Reply:
I have the feeling that having huge json responses in code are too much noise that distract.

Reply:
Yes, good point.